variants/esp32-rpc/components/foo/foo.toit

```diff
@@ -0,0 +1,40 @@
+
+import expect
+import monitor
+
+TYPE ::= 100  // Don't overlap with system messages.
+EXTERNAL_PID ::= 0
+
+main:
+  print "starting"
+  handler := MessageHandler
+  set_system_message_handler_ (TYPE + 1) handler
+  expect.expect (Process.current.id != EXTERNAL_PID)
+
+  test handler #[1]
+  test handler #[1, 2, 3, 4]
+  test handler #[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+  test handler (ByteArray 3: it)
+  test handler (ByteArray 319: it)
+  test handler (ByteArray 3197: it)
+  test handler (ByteArray 31971: it)
+  test handler #[99, 99]
+
+test handler/MessageHandler data/ByteArray:
+  copy := data.copy  // Data can be neutered as part of the transfer.
+  process_send_ EXTERNAL_PID TYPE data
```

Comment:
Maybe it is worth abstracting a little bit over this. If you end up with a symbolic name for external message handlers then having a wrapper, so you can add a helper for sending messages and maybe for receiving them with a lambda instead of a full SystemMessageHandler_ would be nice.

You could have a `--copy` flag for the sending (default to true) like in the Artemis package: https://github.com/toitware/toit-artemis/blob/main/src/artemis.toit#L339.